jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -268,7 +268,134 @@ private static StringBuilder encodePath(StringBuilder buf, String path, int offs
 
         return buf;
     }
-    
+
+    /**
+     * Encode a raw URI String and convert any raw spaces to
+     * their "%20" equivalent.
+     *
+     * @param str input raw string
+     * @return output with spaces converted to "%20"
+     */
+    public static String encodeSpaces(String str)
+    {
+        return StringUtil.replace(str, " ", "%20");
+    }
+
+    /**
+     * Encode a raw String and convert any specific characters to their URI encoded equivalent.
+     *
+     * @param str input raw string
+     * @param charsToEncode the list of raw characters that need to be encoded (if encountered)
+     * @return output with specified characters encoded.
+     */
+    @SuppressWarnings("Duplicates")
+    public static String encodeSpecific(String str, String charsToEncode)
+    {
+        if ((str == null) || (str.length() == 0))
+            return null;
+
+        if ((charsToEncode == null) || (charsToEncode.length() == 0))
+            return str;
+
+        char[] find = charsToEncode.toCharArray();
```

Comment:
it the cost of creating the char array better than just calling charsToEncode.contains or indexOf ?

Reply:
using `indexOf` will create 2 char arrays per call, this is actually better, and the JIT loves to optimize it.